### PR TITLE
[#156913707] Add credit card details

### PR DESCRIPTION
**Pull request by fgiobergia**
When adding a new credit card to the available payment methods, the card details (card holder, pan, expiration date, csc) are to be entered by the user. 

This PR adds the screen that collects this kind of information (screenshot below)

**EDIT**: updated screenshot on @matteodesanti 's instructions: black background for mastercard and maestro, darker placeholder text

Feedback appreciated :)

### CODE REVIEW

**Comment by digitalcitizenship** [comment]
## Affected stories

  * 🌟 [#156913707](https://www.pivotaltracker.com/story/show/156913707): Come CIT voglio poter aggiungere e salvare una nuova carta di credito nel Portafoglio

<p align="right">
  Generated by :no_entry_sign: <a href="http://github.com/danger/danger-js/">dangerJS</a>
</p>

File: ts/screens/wallet/AddCardScreen.tsx [hunk 1]
```diff
@@ -0,0 +1,260 @@
...
+
+  /**
+   * method to modify the formatting of the expiration date
+   * it currently adds a "/" after the month (if a valid
+   * value is entered) and converts 4-digits years in their
+   * 2-digits versions
+   * This is a draft version -- the exact behavior should be
+   * defined somewhere in order for it to be implemented
+   */
+  public onExpirationDateChange = (value: string) => {
```
Comment (cloudify) [inline ts/screens/wallet/AddCardScreen.tsx]: This logic should be moved somewhere else (e.g. `/utils/input.ts`) and throughly tested.

File: ts/screens/wallet/AddCardScreen.tsx [hunk 2]
```diff
@@ -0,0 +1,260 @@
...
+      this.setState({
+        expirationDate: filteredValue.length > 0 ? some(filteredValue) : none
+      });
+    }
+  };
+
+  // update pan so as to add a space every 4 digits
+  // the exact behavior should be defined before
+  // being implemented
+  public onPanChange = (value: string) => {
```
Comment (cloudify) [inline ts/screens/wallet/AddCardScreen.tsx]: same for this logic

**Comment by matteodesanti** [comment]
Great job @fgiobergia ! A couple of fixes: the mastercard and maestro cards should have a dark background https://invis.io/48GKAY7S9AZ#/291923321_2-4-2_-_Dati_Carta
Probably the color of the placeholder text should be a slightly darker.
:)

File: ts/utils/__tests__/input.ts [hunk 3]
```diff
@@ -0,0 +1,80 @@
+import { fixExpirationDate, fixPan } from "../input";
```
Comment (cloudify) [inline ts/utils/__tests__/input.ts]: test files should be named with `.test.ts` extension

File: ts/screens/wallet/AddCardScreen.tsx [hunk 4]
```diff
@@ -0,0 +1,222 @@
...
+  Text,
+  View
+} from "native-base";
+import * as React from "react";
+import { FlatList, Image, ScrollView, StyleSheet } from "react-native";
+import { NavigationScreenProp, NavigationState } from "react-navigation";
+
+import I18n from "../../i18n";
+import ROUTES from "../../navigation/routes";
+
```
Comment (fpersico) [inline ts/screens/wallet/AddCardScreen.tsx]: Is this empty line (and the others) needed?

File: ts/screens/wallet/AddCardScreen.tsx [hunk 5]
```diff
@@ -0,0 +1,222 @@
...
+    flex: 0
+  }
+});
+
+const CARD_LOGOS_COLUMNS = 4;
+
+export class AddCardScreen extends React.Component<Props, State> {
+  constructor(props: Props) {
+    super(props);
+    this.state = {
```
Comment (fpersico) [inline ts/screens/wallet/AddCardScreen.tsx]: Remember we are using redux form

Comment (fgiobergia) [reply ts/screens/wallet/AddCardScreen.tsx]: Gotcha, I will be using it when working on the (upcoming) "card adding" process :)

**Comment by cloudify** [comment]
@fgiobergia I think we can merge this

**Pull request merged**